### ISSUE TITLE
Worksheet -> [[Text]] function?

### ISSUE THREAD

**Issue by mb21** [qrilka/xlsx#21]
This is not really urgent, but I was thinking of adding xlsx support (in addition to csv) to the [pandoc-placetable filter](https://github.com/mb21/pandoc-placetable), so I'd need a `Worksheet -> [[Text]]` function.

I guess I could come up with the fold `Map (Int, Int) Cell -> [[Cell]]` myself somehow (although there might be a neater way using Lens.Traversal?), but how about a `Cell -> Text` function that preserves some styling for numbers (like number of significant digits, currency, scientific notation etc.)?

**Comment by qrilka** [comment]
Currently xlsx does not support cell formatting - `Styles` is just a wrapper around a bytestring. I suppose it is required for your filter, right?

**Comment by mb21** [comment]
Yeah, I mean exporting to csv from a spreadsheet program usually preserves that kind of formatting... so without that I'm kind of hesitating to add xlsx as a dependency... thanks anyway :)

**Comment by qrilka** [comment]
I'll try to look into that area but no particular promises at the moment

**Comment by olafklinke** [comment]
FWIW, I happened to implement something along these lines, because the [xls](https://hackage.haskell.org/package/xls) package has this and I wanted a unified Megaparsec parser for both xlsx and xls. Here is the code:
```
import Control.Arrow (first)
import Data.Function (on)
import Data.List (groupBy)
import Data.Map.Strict (toAscList)
cellRows :: CellMap -> [[Cell]]
cellRows = fmap (fillBlanks . fmap (first snd)) . groupBy ((==) `on` (fst.fst)) . toAscList where
    fillBlanks :: [(Int,Cell)] -> [Cell]
    fillBlanks = go 0
    blankCell = Cell Nothing Nothing Nothing Nothing
    go :: Int -> [(Int,Cell)] -> [Cell]
    go _ [] = []
    go i ((j,c):cs) = replicate (j-i-1) blankCell ++ (c : go j cs) 
```

**Comment by qrilka** [comment]
@olafklinke could we worth creating a PR if you have time for it

**Comment by olafklinke** [comment]
> @olafklinke could we worth creating a PR if you have time for it

IMHO this would actually better be part of a separate package, see [here](https://github.com/qrilka/xlsx/issues/15#issuecomment-2117537356).
